### ISSUE TITLE
A kind of defunct process for timeout mutant

### ISSUE THREAD

**Issue by unlight** [stryker-mutator/stryker#289]
If mutant was not killed properly (infinite loop => timeout), a some skind of defunct process is sitting in the process list.

Maybe related to stryker-karma-runner.

Modules:
```
    "karma-jasmine": "1.1.0",
    "karma-nightmare": "0.4.6",
    "karma-webpack": "2.0.3",
    "stryker": "0.6.0",
    "stryker-api": "0.5.0",
    "stryker-jasmine": "0.2.0",
    "stryker-karma-runner": "0.4.0",
```
ENV:
win 10 x64, node 7.9

How to repeat:
```
git clone https://github.com/unlight/stryker-miniq-demo
cd stryker-miniq-demo
npm i
npm run stryker
```

Issue is not reproduced if there are no timeout mutants,

**Comment by nicojs** [comment]
@unlight thanks for the detailed report and reproduction steps!

I have a win10 x64 machine as well, so i can probably reproduce this issue. 

>Some skind of defunct process is sitting in the process list.

What is the name of the hanging process? Is it still there after Stryker is done? I to see a lot of processes being started, but they eventually get killed after i see:

```
[2017-04-26 19:13:56.236] [INFO] Stryker - Done in 39 seconds.
```

**Comment by unlight** [comment]
> What is the name of the hanging process? 

I'm using `karma-nightmare` as browser, which uses Electron under the hood, so the process name is
`electron.exe`

> Is it still there after Stryker is done?

Yes

**Comment by avassem85** [comment]
Is this not related to https://github.com/bokuweb/karma-nightmare/issues/8?

**Comment by unlight** [comment]
I don't think so.
Because when comment code which mutates to timeout mutant, there is no issues - all process are killed.

**Comment by nicojs** [comment]
Yeah, i'm looking into it. A timeout results in stryker sending a SIGTERM using [process.kill](https://nodejs.org/api/child_process.html#child_process_child_kill_signal). I'm trying to figure out what the normal way is of shutting down karma. For example if you're pressing ctrl+c when karma is running. We should mimic that behavior i think.

**Comment by nicojs** [comment]
I'm not sure if we can do anything about this issue. I can reproduce this issue without Stryker.

1. Create an infinite loop in a regular test 
2. Start karma the normal way 
3. Wait for this message (or simular)
```
Electron 1.6.6 (Windows 10 0.0.0): Executed 0 of 3 SUCCESS (0 secs / 0 secs)
Electron 1.6.6 (Windows 10 0.0.0) ERROR
  Disconnected, because no message in 10000 ms.
```
4. Kill the process (ctrl+c)

Open your task manager. The electron process hangs.

**Comment by simondel** [comment]
@nicojs Can we close this? Stryker has been updated a lot in the past year and we have reworked the sandboxes/childprocesses.
